python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -122,12 +124,20 @@ def _load_trials(self) -> List[Trial]:
         with self._fs.open_input_stream(self._experiment_json_fs_path) as f:
             experiment_state = json.loads(f.readall(), cls=TuneFunctionDecoder)
 
+        experiment_fs_path = Path(self._experiment_fs_path)
+
         trials = []
         trial_states = experiment_state["trial_data"]
         for trial_json_state, trial_runtime_metadata in trial_states:
             trial = Trial.from_json_state(trial_json_state, stub=True)
             trial.restore_run_metadata(trial_runtime_metadata)
-            # TODO(justinvyu): [handle_moved_storage_path]
+
+            new_storage = copy.copy(trial.storage)
+            new_storage.storage_fs_path = experiment_fs_path.parent.as_posix()
+            new_storage.storage_filesystem = self._fs
+            new_storage.experiment_dir_name = experiment_fs_path.name
+            trial.set_storage(new_storage)
```

Comment:
nit: Would it be worthwhile to combine this into a utility since the same logic is used in both places?

Reply:
I think the 2 cases are a bit too different for a common utility to be that useful